core/node/base_token_adjuster/src/base_token_ratio_persister.rs

```diff
@@ -157,36 +120,120 @@ impl BaseTokenRatioPersister {
         (base_fee_per_gas, priority_fee_per_gas)
     }
 
+    async fn retry_update_ratio_on_l1(&self, new_ratio: BaseTokenAPIRatio) -> anyhow::Result<()> {
+        let Some(l1_params) = &self.l1_params else {
+            return Ok(());
+        };
+
+        let start_time = millis_since_epoch();
+        let max_attempts = self.config.l1_tx_sending_max_attempts;
+        let sleep_duration = self.config.l1_tx_sending_sleep_duration();
+        let mut prev_base_fee_per_gas: Option<u64> = None;
+        let mut prev_priority_fee_per_gas: Option<u64> = None;
+        let mut last_error = None;
+        for attempt in 0..max_attempts {
+            let (base_fee_per_gas, priority_fee_per_gas) =
+                self.get_eth_fees(l1_params, prev_base_fee_per_gas, prev_priority_fee_per_gas);
+
+            let result = self
+                .update_ratio_on_l1(l1_params, new_ratio, base_fee_per_gas, priority_fee_per_gas)
+                .await;
+
+            match result {
+                Ok(x) => {
+                    tracing::info!(
+                "Updated base token multiplier on L1: numerator {}, denominator {}, base_fee_per_gas {}, priority_fee_per_gas {}",
+                new_ratio.numerator.get(),
+                new_ratio.denominator.get(),
+                base_fee_per_gas,
+                priority_fee_per_gas
+            );
```

Comment:
nit: please align code

Reply:
Good point, let me realign it. This is actually what `zk fmt` does by default.